Sources/Workspace/Workspace.swift

```diff
@@ -516,6 +516,18 @@ public class Workspace {
             RepositoryPackageConstraint(
                 container: dependency.repository, requirement: requirement))
 
+        // Add unversioned constraint for edited packages.
+        let currentManifests = loadDependencyManifests(rootManifests: rootManifests, engine: engine)
+        for (externalManifest, managedDependency) in currentManifests.dependencies {
+            switch managedDependency.state {
+            case .checkout: continue
+            case .unmanaged, .edited: break
+            }
+            let specifier = RepositorySpecifier(url: externalManifest.url)
+            let dependencies = externalManifest.package.dependencyConstraints()
+            constraints += [RepositoryPackageConstraint(container: specifier, requirement: .unversioned(dependencies))]
```

Comment:
^^ this bit looks to be in 3 different places in Workspace.swift with this modification, so yeah - totally makes sense to refactor that into a method. Will do that.